internal/vmcompute/vmcompute.go

```diff
@@ -544,6 +545,26 @@ func HcsGetServiceProperties(ctx gcontext.Context, propertyQuery string) (proper
 	})
 }
 
+func HcsFormatWritableLayerVhd(ctx gcontext.Context, handle uintptr) error {
+	var err error
+	ctx, span := trace.StartSpan(ctx, "HcsFormatWritableLayerVhd")
+	defer span.End()
+	defer func() {
+		if err != nil {
+			span.AddAttributes(trace.StringAttribute("error", err.Error()))
+		}
+		oc.SetSpanStatus(span, err)
+	}()
+
+	return execute(ctx, timeout.SyscallWatcher, func() error {
```

Comment:
Does this syscall watcher thing still time out requests? In this case it's absolutely not safe to do so because the callee may still be using `handle` but the caller may close the handle after this function returns.

Reply:
(It's probably never actually safe)

Reply:
That is the pattern that is used in almost all the system calls that we use in hcsshim. But you are right that it can close the handle while the system call might still be using it. How else can we make sure that the hcsshim doesn't get stuck inside a system call if it is not making any progress?

Reply:
We can't, and IMO we shouldn't. We should fix the platform bugs that cause such calls to get stuck.

It would be interesting to look at recent production data to see if this ever actually happens anymore.

Reply:
I guess then it would be better if we merge this PR now and then create another PR just to remove this Syscallwatcher from all System calls?

Reply:
I think the philosophy behind this was that if we hit the timeout, something is seriously wrong, and it's fine for the callee to blow up if it actually tries to access the handle again.

That said, we seem to only hit this today for HcsTerminateComputeSystem. I think in most cases it could be removed, as HCS reliability has improved since we first did this.

Reply:
@kevpar @jstarks So do we want to keep this in this change or not?

Reply:
I think it's probably fine to leave in for now, since it's presumably already an issue with our other calls.

Reply:
Actually, I don't think this function should be in here, or defined with this pattern of using `execute` as a wrapper, but for different reasons. This file looks to be all (or mostly) functions that do RPC calls into HCS to function. This function isn't exported from vmcompute.dll, and doesn't rely on calling into HCS to function. I think we should just add a new zsyscall file in `internal/hcs`, and call `hcsFormatWritableLayerVhd` directly from `CreateNTFSVHD`.

Reply:
We don't need a span wrapping this function either.

Reply:
Fixed. Please check again.